R/analyse_SAR.CWOSL.R

```diff
@@ -220,27 +236,95 @@ analyse_SAR.CWOSL<- function(
   ...
 ) {
 
+
+  if (is.null(OSL.component)) {
+
+    if(missing("background.integral.min")){
+      stop("[analyse_SAR.CWOSL()] No value set for 'background.integral.min'!")
+    }
+
+    if(missing("background.integral.max")){
+      stop("[analyse_SAR.CWOSL()] No value set for 'background.integral.max'!")
+    }
+
+    ##make life easy
+    if(missing("signal.integral.min")){
+      signal.integral.min <- 1
+      warning("[analyse_SAR.CWOSL()] 'signal.integral.min' missing, set to 1", call. = FALSE)
+    }
+
+    if(missing("signal.integral.max")){
+      signal.integral.max <- 2
+      warning("[analyse_SAR.CWOSL()] 'signal.integral.max' missing, set to 2", call. = FALSE)
+    }
+
+  } else {
+
+    # DM: If a component argument is given, no integration intervals are necessary
+    # But to prevent later errors, we give them NA values
+    signal.integral.min <- NA
+    signal.integral.max <- NA
+    background.integral.min <- NA
+    background.integral.max <- NA
+
+    # DM: The definition of a CW-OSL curve type is mandatory for decomposed data sets (because of data consistency).
+    # If none is given, take the one used at OSLdecomposition::RLum.OSL_decomposition()
+    if (is.null(CWcurve.type)) {
+      if ("DECOMPOSITION" %in% names(object)) {
+        CWcurve.type <- object$DECOMPOSITION$parameters$record_type
+
+      } else {
+        stop("[analyse_SAR.CWOSL()] Neither an argument CWcurve.type is defined nor contains the object a slot $DECOMPOSITION")}}
+
+    # DM: R session crashes if we plot OSLdecomposed data sets
+    if (plot) {
+      plot <- FALSE
+      warning("[analyse_SAR.CWOSL()] Plot output for OSLdecomposition data sets is currently no available")}
+  }
+
 # SELF CALL -----------------------------------------------------------------------------------
 if(is.list(object)){
 
-  ##make life easy
-  if(missing("signal.integral.min")){
-    signal.integral.min <- 1
-    warning("[analyse_SAR.CWOSL()] 'signal.integral.min' missing, set to 1", call. = FALSE)
-  }
+  # DM: The the RLum.XXX functions of the package OSLdecomposition add list items which are not RLum.Analysis
+  # They may also rename @RecordType values, so some list items may miss any 'record_type' record
+  for (j in length(object):1) {
```

Comment:
Hm, this is possible, but basically breaks the `RLum-class` structure. It just does not flag this here, because we do not run integrity tests on the RLum-objects for performance reasons. What you create with your function is an `RLum.Results` object. You should either use this or  we have to create a new `RLum.Data-class`.